community/ndp/transport-socket/src/main/java/org/neo4j/ndp/transport/socket/ChunkedInput.java

```diff
@@ -193,7 +193,7 @@ private void ensure( int numBytes ) throws IOException
 
     private void ensureChunkAvailable() throws IOException
     {
-        if ( currentChunk == null || currentChunk.readableBytes() == 0 )
+        while ( currentChunk == null || currentChunk.readableBytes() == 0 )
```

Comment:
The trick is gonna be in SocketProtocolV1, and remembering that there's two levels of fragmentation - we split messages into chunks, but the network might then split those packets even further, so you get partial chunks. For that reason, the code in SocketProtocolV1 reads the chunk headers and does the actual de-chunking, appending each little sliver of a chunk it gets off the network into the ChunkedInput. I think (though I wasn't able to prove) that there's a code path in the dechunking there that would lead to appending an empty buffer.

So I'm not sure of the exact code path that caused SocketProtocolV1 to append an empty buffer to ChunkedInput, but this change resolved the large-scale test error, and is covered by the tiny unit test.